database/sql/src/main/scala/cromwell/database/slick/tables/MetadataEntryComponent.scala

```diff
@@ -177,6 +177,33 @@ trait MetadataEntryComponent {
     } yield metadataEntry).sortBy(_.metadataTimestamp)
   }
 
+  /**
+    * Returns metadata entries containing outputs metadata key for given workflow ids
+    */
+  def metadataEntriesForOutputs(workflowIdList: Set[String], outputMetadataKey: List[String]) = {
+    for {
+      metadataEntry <- metadataEntries
+      if metadataEntryHasWorkflowId(metadataEntry, workflowIdList)
+      if metadataEntryHasMetadataKeysLike(metadataEntry, outputMetadataKey, List.empty[String])
+    } yield metadataEntry
+  }
+
+  /**
+    * Returns subworkflow ids for a root workflow id
+    */
+  def subworkflowIdsForRootWorkflow(rootWorkflowId: String) = {
```

Comment:
I feel like the summary entry table could serve us better when computing these, because it's shorter... any reason not to?

Reply:
It might be possible that by the time we start this process of gathering intermediate files, the metadata table might not have been summarized and so the summary table might not have the necessary details. _(very less probability but it could still happen I think)_

Reply:
I am worried about performance on big `METADATA_ENTRY` tables. The only index we have is on `WORKFLOW_EXECUTION_UUID` which dose not help with looking up by `metadataKey` or `metadataValue`.